.travis.yml

```diff
@@ -1,48 +1,59 @@
+# After changing this file, check it on:
+#   http://lint.travis-ci.org/
 language: python
-env:
-  - TEST_PYCRYPTO="true" TEST_GMPY="true"
-  - TEST_PYCRYPTO="false" TEST_GMPY="true"
-
-# Note travis times out with the following:
-#  - TEST_PYCRYPTO="true" TEST_GMPY="false"
+sudo: false
 
 # In order to test with and without some libraries, we explicitly
 # install deps instead of installing from the requirements file
 
-python:
-  - "3.3"
-  - "3.4"
-  - "3.5"
-  - "3.6"
+matrix:
+  include:
+    - python: 3.3
+    - python: 3.4
+    - python: 3.5
+      env:
+        - TEST_PYCRYPTO=1
+    - python: 3.6
+
+addons:
+  apt:
+    packages:
+    - libatlas-dev
+    - libatlas-base-dev
+    - liblapack-dev
+    - gfortran
+    - libgmp-dev
+    - libmpfr-dev
 
 before_install:
+  - export PYTHONFAULTHANDLER=1
+  - export PATH=/usr/lib/ccache:$PATH
   - uname -a
   - free -m
   - df -h
   - ulimit -a
-  - sudo apt-get update -qq
-  - sudo apt-get install -qq python3-crypto
-  - sudo apt-get install -qq libgmp-dev libmpfr-dev
-  - sudo apt-get install -qq libgmp3-dev libgmp3c2 libmpfr4 libmpc-dev libmpc2
-  - pip install --upgrade pip
-  - pip install --upgrade nose
-  # Install gmpy2 dependency
-  - if [[ $TEST_GMPY == "true" ]]; then
-      wget ftp://ftp.gnu.org/gnu/mpc/mpc-1.0.3.tar.gz;
-      tar xzvf mpc-1.0.3.tar.gz;
-      cd mpc-1.0.3;
-      ./configure;
-      make;
-      sudo make install;
-      cd ..;
-      pip install gmpy2==2.0.5;
-    fi
-  # End install gmpy2 dependencies
-  - if [[ $TEST_PYCRYPTO == "true" ]]; then
-      pip install pycrypto==2.6.1;
-    fi
-install:
-  - pip install numpy
-  - pip install click
-script:  nosetests -v
+  - mkdir builds
+  - pushd builds
+  # Install gmpy2 dependencies
+  - mkdir -p $HOME/.local
+  - wget ftp://ftp.gnu.org/gnu/mpc/mpc-1.0.3.tar.gz
+  - tar xzvf mpc-1.0.3.tar.gz
+  - pushd mpc-1.0.3
+  - ./configure --prefix=$HOME/.local
+  - make
+  - make install
+  - popd
+  - export CPATH=$HOME/.local/include
+  - export LIBRARY_PATH=$HOME/.local/lib
+  - export LD_LIBRARY_PATH=$HOME/.local/lib
+  - travis_retry pip install gmpy2
+  - if [ "${TESTMODE}" == "full" ]; then pip install coverage; fi
```

Comment:
Coverage does slow things down a bit, so I wanted to just run it for python 3.5.